clang/include/clang/Basic/DiagnosticDriverKinds.td

```diff
@@ -442,6 +442,9 @@ def warn_drv_deprecated_arg : Warning<
 def warn_drv_deprecated_arg_no_relaxed_template_template_args : Warning<
   "argument '-fno-relaxed-template-template-args' is deprecated">,
   InGroup<DeprecatedNoRelaxedTemplateTemplateArgs>;
+def warn_drv_deprecated_arg_ofast : Warning<
+  "argument '-Ofast' is deprecated; use '-O3', possibly with '-ffast-math'">,
```

Comment:
Maybe worth using a few more words? E.g.
`argument '-Ofast' is deprecated; use '-O3' for standard optimization, or '-O3 -ffast-math' if non-standard "fast-math" was desired`

I'm not sure if that's too long.

Reply:
We could shorten slightly it to:
`'-Ofast' is deprecated; use '-O3' for conforming optimizations or '-O3 -ffast-math' for nonconforming optimizations"`

Reply:
I think in this diagnostic it's more important to inform the user that he should replace `-Ofast` with `-O3 -ffast-math` in order to just circumvent the deprecation and keep the current behavior, and this should be clear. Throwing the option of using just `O3` as equally valid is a bit misleading, but we can, as a note, inform the user that it's possible he made the wrong choice in the first place, and advise a switch to `O3`.

Reply:
I think @mizvekov raised a good point, so I tried to take it into account.
@jyknight @AaronBallman What do you think of the new wording (`use '-O3 -ffast math' instead, or just '-O3' if you don't intend to enable non-conforming optimizations`)?

Reply:
> use '-O3 -ffast math' instead, or just '-O3' if you don't intend to enable non-conforming optimizations

This message LGTM.

Reply:
How about a slight rewording?

`use '-O3 -ffast math' for the same behavior, or '-O3' to enable conforming optimizations only`

(I mostly want to drop "just" and to make it more clear that `-Ofast` was non-conforming because of `-ffast-math`.)

Reply:
Switched to Aaron's wording